Code/Framework/AzToolsFramework/Tests/PaintBrush/PaintBrushBaseTests.cpp

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/PaintBrush/PaintBrush.h>
+#include <AZTestShared/Math/MathTestHelpers.h>
+#include <Tests/PaintBrush/MockPaintBrushNotificationHandler.h>
+
+/*
+This set of unit tests validates that the basic PaintBrush functionality and notifications work:
+- Begin/End Paint Mode
+- Begin/End Brush Stroke
+- Use Eyedropper
+*/
+
+namespace UnitTest
+{
+    class PaintBrushTestFixture : public ScopedAllocatorSetupFixture
+    {
+    public:
+
+        // Some common default values that we'll use in our tests.
+        const AZ::EntityComponentIdPair EntityComponentIdPair{ AZ::EntityId(123), AZ::ComponentId(456) };
+        const AZ::Vector3 TestBrushCenter{ 10.0f, 20.0f, 30.0f };
+        const AZ::Vector3 TestBrushCenter2d{ 10.0f, 20.0f, 0.0f };  // This should be the same as TestBrushCenter, but with Z=0
+        const AZ::Color TestColor{ 0.25f, 0.50f, 0.75f, 1.0f };
+
+        // Useful helpers for our tests
+        AzToolsFramework::PaintBrushSettings m_settings;
+    };
+
+    TEST_F(PaintBrushTestFixture, TrivialConstruction)
+    {
+        // We can construct / destroy a PaintBrush without any errors.
+
+        AzToolsFramework::PaintBrush paintBrush(EntityComponentIdPair);
+    }
+
+    TEST_F(PaintBrushTestFixture, BeginAndEndPaintModeMustBePairedCorrectly)
+    {
+        // BeginPaintMode() / EndPaintMode() can only be called in the correct order in pairs.
+        // They will error if they're called in the wrong order or more than once in a row.
+
+        AzToolsFramework::PaintBrush paintBrush(EntityComponentIdPair);
+
+        // End before Begin should assert.
+        AZ_TEST_START_TRACE_SUPPRESSION;
```

Comment:
Hmmm, thanks for pointing me at it! I've got mixed feelings about using it. I like that you can make sure you're not getting a false positive from an unexpected assert, but I don't like that it's specifically testing the wording of the assert, since that's something that should be able to change without breaking unit tests.

My inclination is to _not_ use it here, since these are some pretty trivial assert tests that would be difficult to fail unexpectedly, but if you feel strongly about it I can switch them.

Reply:
It's totally fine to leave as-is I just wanted to mention it as an option (might be useful in another context in future if you want to check something specific). Agreed it's not perfect but for asserts that's about the only 'unique' property they have (the associated string/text).